### PR TITLE
[INFRA] Automate tag and files update

### LINKED ISSUE #47: [INFRA] Automate tag and files update
Currently (in v0.2.1), the version must be updated prior and after releasing (by creating a PR). This is painful and error prone
The git tag is created only when the GitHub release is published.

I suggest we do something like for the bpmn-visualization TypeScript library.
We provide a GitHub workflow that will do everything for us:
- It takes as an input the version to release.
- it updates the files with the new version and commit
- then it creates the related tag
- then it updates the files with the development version and commit
- finally, it push the changes and the tag

If possible, the development version should be automatically computed by the workflow to avoid human error

### Tasks to do

Add a Github workflow triggered by `workflow_dispatch` (manual run)
- [x] pass the new version
- [x] update the files containing the version accordingly (see MAINTAINERS.md)
- [x] create the git tag
- [x] set the development version accordingly (see MAINTAINERS.md)
- [x] push all commits and the tag

Documentation
- [x] update MAINTAINERS.md to explain the new release process

### Notes
- remember that the GH_TOKEN has readonly permissions by default
- check how we proceed in the bpmn-visualization-js repo
- don't forget to use the playground repository for testing

### PR DESCRIPTION
**Pull request by csouchet**
Add new workflow for the release: `release.yml` including the following steps (in order):
  - unprotect the branch
  - generate the new version with https://github.com/marketplace/actions/create-new-semantic-version (based on the latest git tag)
  - update the README.md and DESCRIPTION files with the new version
  - commit the changes
  - tag, then push the commit and tag
  - update the DESCRIPTION file with the new development version
  - commit and push it
  - protect the branch again

Closes #47 

### Test
See this PR to see the tests with `patch`, `minor` and `major` version: https://github.com/process-analytics/github-actions-playground/pull/173

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          token: ${{ secrets.GH_RELEASE_TOKEN }}
+
+      - name: Config git
+        run: |
+          git config --local user.email "62586190+process-analytics-bot@users.noreply.github.com"
+          git config --local user.name "process-analytics-bot"
+          git config pull.rebase true
+
+      - name: Checkout master
+        run: git checkout master && git pull --tag
```
Comment (tbouffard) [inline .github/workflows/release.yml:37]: why not using the checkout action here?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      - name: Commit
+        run: |
+          git add README.md
+          git add DESCRIPTION
+          git commit -m "[RELEASE] Prepare version for release"
+          git push
+
+      - name: Tag ${{ steps.next_version.outputs.version }}
+        run: |
+          git tag -a v${{ steps.next_version.outputs.version }} -m "[RELEASE] ${{ steps.next_version.outputs.version }}"
```
Comment (tbouffard) [inline .github/workflows/release.yml:70]: why did you choose to create an annotated git tag here?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      - name: Push Version
+        run: git push && git push --tags
+
+      - name: Branch Protection Bot - Reenable "include administrators" branch protection
+        uses: benjefferies/branch-protection-bot@1.0.7
+        if: always()  # Force to always run this step to ensure "include administrators" is always turned back on
+        with:
+          access_token: ${{ secrets.GH_RELEASE_TOKEN }}
+          enforce_admins: true
+          branch: master
```
Comment (tbouffard) [inline .github/workflows/release.yml:81]: ⚠️ there is no `master` branch in this repository, but a `main` branch

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      - run: |
+          echo "New version type: ${{ github.event.inputs.type }}"
+
+      - name: Branch Protection Bot - Temporarily disable "include administrators" branch protection
+        uses: benjefferies/branch-protection-bot@1.0.7
+        if: always()
+        with:
+          access_token: ${{ secrets.GH_RELEASE_TOKEN }}
+          enforce_admins: false
+          branch: master
```
Comment (tbouffard) [inline .github/workflows/release.yml:23]: ⚠️ there is no `master` branch in this repository, but a `main` branch

**Review by tbouffard — COMMENTED** [review]
❓ if we automate the files update, we should update the release process documentation as well.
See also #146

File: .github/workflows/release.yml (hunk 2 above)
Comment (csouchet) [reply .github/workflows/release.yml]: ℹ️ To add a message on the tag, it's the only way according the documentation : https://git-scm.com/book/en/v2/Git-Basics-Tagging.
According https://git-scm.com/docs/git-tag, if `-m <msg>` or `-F <file>` is given and `-a`, `-s`, and `-u <keyid>` are absent, `-a` is implied.

So I can remove it, but it is the same result 🙂

Comment (tbouffard) [reply .github/workflows/release.yml]: fine

**Review by tbouffard — APPROVED** [review]
LGTM

File: MAINTAINERS.md [hunk 5]
```diff
@@ -38,37 +27,60 @@ if some files require changes. The PR/commit message should be `[RELEASE] Set ve
...
+
+___
 
-- Open the draft release note in [github releases](https://github.com/process-analytics/bpmn-visualization-R/releases)
-- Verify the tag version as the same as the release note.
-- Published the release **only when you are done** with the release content.
+### Manual procedure
 
-### Use a development version
+⚠️⚠️⚠️ _**Only if the GitHub job that publish the npm package fails after manual re-run!**_  ⚠️⚠️⚠️
```
Comment (csouchet) [inline MAINTAINERS.md:53]:
```suggestion
⚠️⚠️⚠️ _**Only if the GitHub job that release this repository fails after manual re-run!**_  ⚠️⚠️⚠️
```

**Pull request merged**